File: dev-docs/snippets/databases/update-operation.ts
import type { JsBaoClient } from "js-bao-wss-client";

// Update an existing operation's definition or access level.
export async function updateOperation(
  client: JsBaoClient,
  databaseId: string,
  name: string,
) {
  // #region example
  const op = await client.databases.updateOperation(databaseId, name, {
    access: "public",
  });
  // #endregion example
  return op;
}

File: dev-docs/snippets/databases/update-operation.swift
import JsBaoClient

// Update an existing operation's definition or access level.
func updateOperation(client: JsBaoClient, databaseId: String, name: String) async throws {
  // #region example
  let op = try await client.databases.updateOperation(
    databaseId: databaseId,
    name: name,
    params: ["access": "public"]
  )
  // #endregion example
  _ = op
}
